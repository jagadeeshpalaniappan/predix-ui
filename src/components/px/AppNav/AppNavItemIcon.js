import styled, { css } from 'styled-components';

// app-nav-item__icon
const AppNavItemIcon = styled.div`
  max-width: 32px;

  ${props => props.empty && css`
    display: block;
    width: var(--px-app-nav-item-icon-size, 2rem);
    height: var(--px-app-nav-item-icon-size, 2rem);
    background-color: var(--px-app-nav-item-background-color--empty, lightgray);
    flex: none;
  `}

  ${props => props.withLabel && css`
    margin-right: 0.33333rem;
  `}

  ${props => props.dropdownIcon && css`
    margin-left: 0.3rem;
    --iron-icon-height: 20px;
    --iron-icon-width: 20px;
  `}
`;
AppNavItemIcon.defaultProps = { className: 'app-nav-item__icon' };
AppNavItemIcon.displayName = 'AppNavItemIcon';

export default AppNavItemIcon;

