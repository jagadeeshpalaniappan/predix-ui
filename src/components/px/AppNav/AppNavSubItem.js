import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const SubItemLabel = styled.p`
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

const SubItem = styled.div`
  user-select: none;
  line-height: 1.33333;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  display: flex;
  align-items: center;
  padding-left: 1rem;
  padding-right: 1rem;
  cursor: pointer;
  height: var(--px-app-nav-subitem-height, 2rem);
  color: var(--px-app-nav-subitem-text-color, darkgray);
  background-color: var(--px-app-nav-subitem-background-color, white);

  &:hover {
    color: var(--px-app-nav-subitem-text-color--hover, darkgray);
    background-color: var(--px-app-nav-subitem-background-color--hover, lightgray);
  }

  ${props => props.selected && css`
    color: var(--px-app-nav-subitem-text-color--selected, white);
    background-color: var(--px-app-nav-subitem-background-color--selected, blue);
    &:hover {
      color: var(--px-app-nav-subitem-text-color--selected, white);
      background-color: var(--px-app-nav-subitem-background-color--selected, blue);
    }
  `}

  ${props => props.collapsed && css`
    color: var(--px-app-nav-subitem-text-color--collapsed, darkgray);
    background-color: var(--px-app-nav-subitem-background-color--collapsed, darkgray);
    padding-left: 3.66667rem;
    &:hover {
      background-color: var(--px-app-nav-subitem-background-color--collapsed-hover, darkgray);
    }
  `}
`;


const AppNavSubItem = ({
  item, collapsed, onClick
}) => (
  <SubItem
    collapsed={collapsed}
    selected={item.selected}
    disabled={item.disabled}
    data-id={item.id}
    onClick={onClick}
  >
    <SubItemLabel>{item.label}</SubItemLabel>
  </SubItem>
);

AppNavSubItem.defaultProps = {
  item: null,
  collapsed: null,
  onClick: null
};

AppNavSubItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string,
    disabled: PropTypes.bool,
    selected: PropTypes.bool,
    label: PropTypes.string,
    icon: PropTypes.string
  }),
  collapsed: PropTypes.bool,
  onClick: PropTypes.func
};

export default AppNavSubItem;
