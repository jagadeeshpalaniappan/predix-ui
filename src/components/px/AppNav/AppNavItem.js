import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styled, { css } from 'styled-components';

import Icon from '../IconSet/Icon';

import AppNavItemIcon from './AppNavItemIcon';

const AppNavItem = styled.div`
  line-height: 1.33333;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  display: flex;
  align-items: center;
  height: var(--px-app-nav-height, 4rem);
  padding-left: var(--px-app-nav-item-padding, 1rem);
  padding-right: var(--px-app-nav-item-padding, 1rem);
  cursor: pointer;
  color: var(--px-app-nav-item-text-color, darkgray);
  background-color: var(--px-app-nav-item-background-color, lightgray);
  --iron-icon-stroke-color: var(--px-app-nav-item-text-color, darkgray);
  user-select: none;
  outline: none;

  &:hover {
    color: var(--px-app-nav-item-text-color--hover, lightgray);
    background-color: var(--px-app-nav-item-background-color--hover, darkgray);
    --iron-icon-stroke-color: var(--px-app-nav-item-text-color--hover, lightgray);
  }

  &:active {
    color: var(--px-app-nav-item-text-color--pressed, darkgray);
    background-color: var(--px-app-nav-item-background-color--pressed, lightgray);
    --iron-icon-stroke-color: var(--px-app-nav-item-text-color--hover, darkgray);
  }

  ${props => props.selected && css`
    color: var(--px-app-nav-item-text-color--selected, darkgray) !important;
    background-color: var(--px-app-nav-item-background-color--selected, white) !important;
    --iron-icon-stroke-color: var(--px-app-nav-item-text-color--selected, darkgray);
    &:before {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      display: block;
      width: 100%;
      height: var(--px-app-nav-item-stripe-size--selected, 0.13333rem);
      background-color: var(--px-app-nav-item-stripe-color--selected, blue);
    }
    ${props.collapsed && css`
      color: var(--px-app-nav-item-text-color--collapsed, darkgray);
      background-color: var(--px-app-nav-item-background-color--collapsed, white);
      &:before {
        top: 0;
        left: 0;
        bottom: 0;
        right: auto;
        height: 100%;
        width: var(--px-app-nav-item-stripe-size--selected, 0.13333rem);
      }
      .app-nav-item__label {
        flex: 1 1 auto;
      }
    `}
  `}

`;
AppNavItem.displayName = 'AppNavItem';
AppNavItem.defaultProps = { className: 'px-app-nav-item' };


const AppNavItemLabel = styled.div`
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: flex;

  ${props => props.onlyShowIcon && css`
    display: none;
  `}
  ${props => props.collapsed && css`
    flex: 1 1 auto;
  `}
  ${props => props.empty && css`
    display: block;
    width: 6.66667rem;
    height: 0.66667rem;
    background-color: var(--px-app-nav-item-background-color--empty, lightgray);
  `}
`;
AppNavItemLabel.displayName = 'AppNavItemLabel';
AppNavItemLabel.defaultProps = { className: 'app-nav-item__label' };


/**
 * AppNav-item component
 */
class AppNavItemComponent extends React.Component {
  _handleClick = (e) => {
    if (!this.props.cancelSelect && this.props.onClick) {
      this.props.onClick(e);
    }
  }

  _shouldComponentUpdate(nextProps) {
    return nextProps.selected !== this.props.selected ||
      nextProps.onlyShowIcon !== this.props.onlyShowIcon;
  }
  render() {
    const {
      label,
      icon,
      id,
      item,
      items,
      selected,
      collapsed,
      emptyIcon,
      emptyLabel,
      withLabel,
      empty,
      dropdown,
      onlyShowIcon
    } = this.props;

    const baseClasses = classnames(
      'px-app-nav-item',
      { selected }
    );

    const itemProps = {
      label, icon, id, items, selected, dropdown, collapsed
    };

    return (
      <AppNavItem
        onClick={() => { this._handleClick(item); }}
        className={baseClasses}
        dropdown={(items)}
        data-id={id}
        data-label={label}
        {...itemProps}
      >
        {icon && !empty && !emptyIcon &&
          <AppNavItemIcon
            withLabel={label}
          >
            <Icon icon={icon} />
          </AppNavItemIcon>
        }

        {label && !emptyLabel && !emptyIcon && !empty &&
          <AppNavItemLabel
            withLabel={withLabel}
            onlyShowIcon={onlyShowIcon}
          >
            {label}
          </AppNavItemLabel>
        }

        {/* emptyIcon */}
        {emptyIcon && <AppNavItemIcon empty withLabel={label} />}

        {/* emptyLabel */}
        {emptyLabel && <AppNavItemLabel empty />}

        {empty && <AppNavItemIcon empty withLabel={label} />}
        {empty && <AppNavItemLabel empty />}

        {dropdown && !onlyShowIcon &&
          <AppNavItemIcon dropdownIcon>
            <Icon icon="px-utl:chevron" size={19} />
          </AppNavItemIcon>
        }
      </AppNavItem>
    );
  }
}

AppNavItemComponent.defaultProps = {
  /**
   * Set to `true` if the item is in a collapsed dropdown.
   */
  collapsed: false,

  /**
   * Cancels events that trigger selection.
   */
  cancelSelect: false,
  /**
   * Used to set the icon for the item.
   */
  icon: null,
  /**
   * A reference the object used to create this nav item.
   */
  item: null,
  /**
   * Used as the label text for the item.
   */
  label: null,
  /**
   * Set to `true` if the item is selected.
   */
  selected: false,
  /**
   * Shows a dropdown arrow to indicate item can be tapped to open a subgroup.
   */
  dropdown: false,
  empty: false,
  /**
   * Shows an empty state outline for icon.
   */
  emptyIcon: false,
  /**
   * Shows an empty state outline for the label.
   */
  emptyLabel: false,
  /**
   * Set to `true` if the item is inside a overflowed dropdown.
   */
  overflowed: false,
  /**
   * Set to `true` if this is a subitem.
   */
  subitem: false,
  onlyShowIcon: false,
  onClick: null,
  fixedWidth: null,
  id: null,
  items: null,
  hideDropdownIcon: false,
  withLabel: false
};

AppNavItemComponent.propTypes = {
  collapsed: PropTypes.bool,
  withLabel: PropTypes.bool,
  cancelSelect: PropTypes.bool,
  icon: PropTypes.string,
  item: PropTypes.shape({
    id: PropTypes.string,
    label: PropTypes.string,
    icon: PropTypes.string
  }),
  items: PropTypes.arrayOf(PropTypes.any),
  id: PropTypes.string,
  label: PropTypes.string,
  selected: PropTypes.bool,
  dropdown: PropTypes.bool,
  empty: PropTypes.bool,
  onlyShowIcon: PropTypes.bool,
  emptyIcon: PropTypes.bool,
  emptyLabel: PropTypes.bool,
  overflowed: PropTypes.bool,
  subitem: PropTypes.bool,
  onClick: PropTypes.func,
  fixedWidth: PropTypes.bool,
  hideDropdownIcon: PropTypes.bool
};

AppNavItemComponent.displayName = 'AppNavItemComponent';

export default AppNavItemComponent;
