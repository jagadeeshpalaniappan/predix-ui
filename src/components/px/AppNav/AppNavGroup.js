import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import AppNavItem from './AppNavItem';
import AppNavSubItem from './AppNavSubItem';

const AppNavGroup = styled.div`
  color: var(--px-base-text-color, #2c404c);
  line-height: 1.33333;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  display: flex;
  height: auto;
  overflow: hidden;

  .px-app-nav-item {
    flex: 1 1 auto;
  }
`;
AppNavGroup.displayName = 'AppNavGroup';
AppNavGroup.defaultProps = {
  className: 'px-app-nav-group'
};

// app-nav-group__dropdown
const AppNavGroupDropdown = styled.div`
  display: block;
  height: calc(100vh - var(--px-app-nav-height, 4rem));
  z-index: 399;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.3);
  ${props => props.opened && css`
    height: auto;
    overflow: auto;
  `}
`;
AppNavGroupDropdown.displayName = 'AppNavGroupDropdown';
AppNavGroupDropdown.defaultProps = {
  className: 'app-nav-group__dropdown'
};

// app-nav-group__dropdown__content
const AppNavGroupDropdownContent = styled.div`
  background-color: var(--px-app-nav-subitem-background-color, white);
  max-width: inherit;
  margin: 1px;
  position: relative;
  top: -1px;
`;
AppNavGroupDropdownContent.displayName = 'AppNavGroupDropdownContent';
AppNavGroupDropdownContent.defaultProps = {
  className: 'app-nav-group__dropdown__content'
};

const AppNavGroupComponent = ({
  item,
  opened,
  selected,
  overflowed,
  collapsed,
  onlyShowIcon,
  emptyIcon,
  onClick
}) => (
  <AppNavGroup>
    <AppNavItem
      dropdown
      onClick={() => onClick && onClick(item)}
      {...item}
      item={item}
      selected={selected}
      overflowed={overflowed}
      collapsed={collapsed}
      onlyShowIcon={onlyShowIcon}
      emptyIcon={emptyIcon}
    />
    <AppNavGroupDropdown opened={opened}>
      <AppNavGroupDropdownContent>
        {item.children && item.children.map(child => (
          <AppNavSubItem
            selected={child.selected}
            onClick={() => onClick && onClick(child, true)}
            key={child.label}
            item={child}
            {...child}
          />
        ))}
      </AppNavGroupDropdownContent>
    </AppNavGroupDropdown>
  </AppNavGroup>
);

AppNavGroupComponent.displayName = 'AppNavGroup';
AppNavGroupComponent.defaultProps = {
  selected: false,
  collapsed: false,
  overflowed: false,
  onClick: null,
  item: null,
  emptyIcon: false,
  opened: false,
  onlyShowIcon: false
};

AppNavGroupComponent.propTypes = {
  onClick: PropTypes.func,
  selected: PropTypes.bool,
  collapsed: PropTypes.bool,
  overflowed: PropTypes.bool,
  item: PropTypes.shape({
    id: PropTypes.string,
    label: PropTypes.string,
    icon: PropTypes.string
  }),
  emptyIcon: PropTypes.bool,
  opened: PropTypes.bool,
  onlyShowIcon: PropTypes.bool
};

export default AppNavGroupComponent;
