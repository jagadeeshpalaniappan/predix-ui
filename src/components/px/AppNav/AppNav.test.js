import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

import AppNav from './';
import AppNavItem from './AppNavItem';

const navItems = [
  {id : "home", label: "Home", icon: "px-fea:home"},
  {id : "settings", label: "Settings", icon: "px-fea:settings"},
  {id : "alert", label: "Alerts", icon: "px-fea:alerts"},
  {
    label: 'Assets',
    id: 'assets',
    icon: 'px-fea:asset',
    children: [
      {
        label: 'Asset #1',
        id: 'a1'
      },
      {
        label: 'Asset #2',
        id: 'a2'
      }
    ]
  }
];

describe('AppNav', () => {

  test('should render', () => {
    const wrapper = shallow(<AppNav items={navItems}/>);
    expect(wrapper.find('.app-nav').exists());
    expect(wrapper.find('[label="Home"]').exists());
    const tree = renderer.create(<AppNav items={navItems}/>).toJSON();
      expect(tree).toMatchSnapshot();
  });

  xtest('should set selected, seletedItem, selectedIndex on "click"', () => {
    let spy = sinon.spy();
    const wrapper = shallow(<AppNav onChange={spy} selected={1} items={navItems}/>);
    const targetItem = wrapper.find('[label="Alerts"]').simulate('click');
    const selectedItem = wrapper.find('.selected');

    expect(wrapper.state().selected).toEqual(2);
    expect(wrapper.state().selectedIndex).toEqual(2);
    expect(wrapper.state().selectedItem.id).toEqual(navItems[2].id);
    expect(selectedItem.exists());
    expect(spy.calledOnce);
  });

  xtest('should set selected, seletedItem, selectedIndex when selected is not null', () => {
    let spy = sinon.spy();
    const wrapper = shallow(<AppNav onChange={spy} selected={1} items={navItems}/>);
    expect(wrapper.state().selectedIndex).toEqual(1);
    expect(wrapper.state().selectedItem.id).toEqual(navItems[1].id);
    expect(selectedItem.exists());
    expect(spy.calledOnce);
  });

  xtest('should work with propForSelect', () => {
    let spy = sinon.spy();
    const wrapper = shallow(<AppNav onChange={spy} selected='home' propForSelect='id' items={navItems}/>);
    //expect(wrapper.state().selectedItem.id).toEqual('home');
    const targetItem = wrapper.find('[label="Alerts"]').simulate('click');
    const selectedItem = wrapper.find('.selected');

    expect(wrapper.state().selected).toEqual('alert');
    expect(wrapper.state().selectedItem.id).toEqual('alert');
    expect(selectedItem.exists());
    expect(spy.calledOnce);
  });

  test('should not update if selected does not change', () =>{
    let spy = sinon.spy();
    const wrapper = shallow(<AppNav onChange={spy} selected={1} items={navItems}/>);
    expect(wrapper.state().selected).toEqual(1);
    wrapper.setProps({selected: 2});
    expect(wrapper.state().selected).toEqual(2);
    wrapper.setProps({selected: 0});
    expect(wrapper.state().selected).toEqual(0);
   //expect(spy.callCount).toEqual(2);

  });

  describe('AppNavItem', () =>{
    it('should render', () =>{
      const wrapper = shallow(<AppNavItem item={navItems[0]}/>);
      expect(wrapper.find('[label="Home"]').exists());
    });

    it('should dispatch click handler', () =>{
      let spy = sinon.spy();
      const wrapper = shallow(<AppNavItem item={navItems[0]} onClick={spy}/>);
      expect(wrapper.find('[label="Home"]').exists());
      wrapper.find('.px-app-nav-item').simulate('click');
      expect(spy.calledOnce);
    });

    it('should render icon', () =>{
      const wrapper = shallow(<AppNavItem {...navItems[0]}/>);
      expect(wrapper.find('[label="Home"]').exists());
      expect(wrapper.find('.px-fea-home').exists());
    });

    it('should render dropdown icon if prop exists', () =>{
      const wrapper = shallow(
      <AppNavItem
        dropdown={true}
        {...navItems[0]}/>
        );
      expect(wrapper.find('[label="Home"]').exists());
      expect(wrapper.find('.app-nav-item__dropdown-icon').exists());
    });

    it('should render empty icon/label if prop exists', () =>{
      const wrapper = shallow(
      <AppNavItem
        emptyIcon={true}
        emptyLabel={true}
        {...navItems[0]}/>
        );
      expect(wrapper.find('[label="Home"]').exists());
      expect(wrapper.find('.app-nav-item__label--empty').exists());
      expect(wrapper.find('.app-nav-item__icon--empty').exists());
    });

    it('should render empty icon/label if empty exists', () =>{
      const wrapper = shallow(
      <AppNavItem
        empty={true}
        {...navItems[0]}/>
        );
      expect(wrapper.find('[label="Home"]').exists());
      expect(wrapper.find('.app-nav-item__label--empty').exists());
      expect(wrapper.find('.app-nav-item__icon--empty').exists());
    });
  });

});
