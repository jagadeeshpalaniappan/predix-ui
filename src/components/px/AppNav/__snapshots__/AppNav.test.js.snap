// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav should render 1`] = `
.c4 {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  vertical-align: middle;
  width: var(--iron-icon-width);
  height: var(--iron-icon-height);
}

.c4 > svg {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  position: relative;
  vertical-align: middle;
}

.c4.hidden {
  display: none;
}

.c3 {
  max-width: 32px;
  margin-right: 0.33333rem;
}

.c7 {
  max-width: 32px;
  margin-left: 0.3rem;
  --iron-icon-height: 20px;
  --iron-icon-width: 20px;
}

.c2 {
  line-height: 1.33333;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: var(--px-app-nav-height,4rem);
  padding-left: var(--px-app-nav-item-padding,1rem);
  padding-right: var(--px-app-nav-item-padding,1rem);
  cursor: pointer;
  color: var(--px-app-nav-item-text-color,darkgray);
  background-color: var(--px-app-nav-item-background-color,lightgray);
  --iron-icon-stroke-color: var(--px-app-nav-item-text-color,darkgray);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
}

.c2:hover {
  color: var(--px-app-nav-item-text-color--hover,lightgray);
  background-color: var(--px-app-nav-item-background-color--hover,darkgray);
  --iron-icon-stroke-color: var(--px-app-nav-item-text-color--hover,lightgray);
}

.c2:active {
  color: var(--px-app-nav-item-text-color--pressed,darkgray);
  background-color: var(--px-app-nav-item-background-color--pressed,lightgray);
  --iron-icon-stroke-color: var(--px-app-nav-item-text-color--hover,darkgray);
}

.c5 {
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c10 {
  line-height: 1.33333;
  -webkit-text-size-adjust: 100%;
  text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1rem;
  padding-right: 1rem;
  cursor: pointer;
  height: var(--px-app-nav-subitem-height,2rem);
  color: var(--px-app-nav-subitem-text-color,darkgray);
  background-color: var(--px-app-nav-subitem-background-color,white);
}

.c10:hover {
  color: var(--px-app-nav-subitem-text-color--hover,darkgray);
  background-color: var(--px-app-nav-subitem-background-color--hover,lightgray);
}

.c6 {
  color: var(--px-base-text-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.33333;
  height: auto;
  box-sizing: border-box;
  position: relative;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c6 .px-app-nav-item {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c8 {
  height: calc(100vh - var(--px-app-nav-height,4rem));
  z-index: 399;
  height: 0;
  overflow: hidden;
}

.c9 {
  background-color: var(--px-app-nav-subitem-background-color,white);
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.3);
  max-width: inherit;
  margin: 1px;
  position: relative;
  top: -1px;
}

.c0 {
  height: var(--px-app-nav-height,4rem);
  background-color: var(--px-app-nav-background-color,whitesmoke);
  box-shadow: var(--px-app-nav-box-shadow,0px 2px 4px var(--px-shadow-navigation,rgba(0,0,0,0.2)));
  display: block;
  width: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c0 .app-nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  height: var(--px-app-nav-height,4rem);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
}

<div
  className="px-app-nav"
>
  <div
    className="app-nav c0"
    onChange={null}
    selected={null}
  >
    <section
      className="app-nav__items c1"
    >
      <div>
        <div
          className="px-app-nav-item c2"
          data-id="home"
          data-label="Home"
          icon="px-fea:home"
          id="home"
          label="Home"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="app-nav-item__icon c3"
          >
            <i
              className="px-icon px-fea-home c4"
              color={null}
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<g id=\\"px-fea-home\\"><path stroke-linejoin=\\"round\\" d=\\"M5.5 16v14.5h21V16m-25-.79L16 2.1l14.5 13.11\\"/><path stroke-linejoin=\\"round\\" d=\\"M12.5 22.5h7v8h-7z\\"/></g>",
                  }
                }
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "display": "block",
                    "height": 32,
                    "pointerEvents": "none",
                    "width": 32,
                  }
                }
                viewBox="0 0 32 32"
              />
            </i>
          </div>
          <div
            className="app-nav-item__label c5"
          >
            Home
          </div>
        </div>
      </div>
      <div>
        <div
          className="px-app-nav-item c2"
          data-id="settings"
          data-label="Settings"
          icon="px-fea:settings"
          id="settings"
          label="Settings"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="app-nav-item__icon c3"
          >
            <i
              className="px-icon px-fea-settings c4"
              color={null}
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "px-fea-settings",
                  }
                }
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "display": "block",
                    "height": 32,
                    "pointerEvents": "none",
                    "width": 32,
                  }
                }
                viewBox="0 0 32 32"
              />
            </i>
          </div>
          <div
            className="app-nav-item__label c5"
          >
            Settings
          </div>
        </div>
      </div>
      <div>
        <div
          className="px-app-nav-item c2"
          data-id="alert"
          data-label="Alerts"
          icon="px-fea:alerts"
          id="alert"
          label="Alerts"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="app-nav-item__icon c3"
          >
            <i
              className="px-icon px-fea-alerts c4"
              color={null}
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<g id=\\"px-fea-alerts\\"><path d=\\"M14.69 3.35L1.7 26.29A1.53 1.53 0 0 0 3 28.58h26a1.53 1.53 0 0 0 1.31-2.29l-13-22.94a1.5 1.5 0 0 0-2.62 0z\\" transform=\\"translate(-1 -2.08)\\" stroke-miterlimit=\\"10\\"/><path stroke-linejoin=\\"round\\" d=\\"M15 17.45h1.56l.78-7.07h-4.68l.78 7.07H15zm-1.4 1.97h2.81v3.37H13.6z\\"/></g>",
                  }
                }
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "display": "block",
                    "height": 32,
                    "pointerEvents": "none",
                    "width": 32,
                  }
                }
                viewBox="0 0 32 32"
              />
            </i>
          </div>
          <div
            className="app-nav-item__label c5"
          >
            Alerts
          </div>
        </div>
      </div>
      <div
        className="px-app-nav-group c6"
      >
        <div>
          <div
            className="px-app-nav-item c2"
            data-id="assets"
            data-label="Assets"
            icon="px-fea:asset"
            id="assets"
            label="Assets"
            onClick={[Function]}
            selected={false}
          >
            <div
              className="app-nav-item__icon c3"
            >
              <i
                className="px-icon px-fea-asset c4"
                color={null}
                style={
                  Object {
                    "height": 32,
                    "width": 32,
                  }
                }
              >
                <svg
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<g id=\\"px-fea-asset\\"><path stroke-linejoin=\\"round\\" d=\\"M25.5 22.25V7.75L13 .5.5 7.75v14.5l13 7.25 12-7.25z\\"/><path stroke-linejoin=\\"round\\" d=\\"M.5 8.22l13 7.25 12-7.25m-12 21.18v-14\\"/><path stroke-linejoin=\\"round\\" d=\\"M6.67 7.99L13 4.31l6.3 3.66-6.36 3.65-6.27-3.63z\\"/></g>",
                    }
                  }
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "display": "block",
                      "height": 32,
                      "pointerEvents": "none",
                      "width": 32,
                    }
                  }
                  viewBox="0 0 32 32"
                />
              </i>
            </div>
            <div
              className="app-nav-item__label c5"
            >
              Assets
            </div>
            <div
              className="app-nav-item__icon c7"
            >
              <i
                className="px-icon px-utl-chevron c4"
                color={null}
                style={
                  Object {
                    "height": 19,
                    "width": 19,
                  }
                }
              >
                <svg
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<g id=\\"px-utl-chevron\\"><path d=\\"M2.4 6.2l5.5 5.5 5.5-5.5\\"/></g>",
                    }
                  }
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "display": "block",
                      "height": 19,
                      "pointerEvents": "none",
                      "width": 19,
                    }
                  }
                  viewBox="0 0 16 16"
                />
              </i>
            </div>
          </div>
        </div>
        <div
          className="app-nav-group__dropdown c8"
        >
          <div
            className="app-nav-group__dropdown__content c9"
          >
            <div
              className="c10"
              data-id="a1"
              onClick={[Function]}
              selected={null}
            >
              <p
                className="app-nav-subitem__label c11"
              >
                Asset #1
              </p>
            </div>
            <div
              className="c10"
              data-id="a2"
              onClick={[Function]}
              selected={null}
            >
              <p
                className="app-nav-subitem__label c11"
              >
                Asset #2
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
